r-package/prep_data/prep_accidents.R

```diff
@@ -0,0 +1,138 @@
+##### Metadata:
+#
+# Data set: Acidentes em Rodovias Federais
+#
+# Source: Polícia Rodoviária Federal (PRF)
+#
+# website: https://portal.prf.gov.br/dados-abertos-acidentes
+#
+# Update frequency: Yearly
+
+
+### Libraries (use any library as necessary)
+
+library(tidyverse)
+library(sf)
+library(geobr)
+library(rvest)
+library(xml2)
+library(glue)
+library(rio)
+library(data.table)
+library(lubridate)
+library(magrittr)
+
+####### Load Support functions to use in the preprocessing of the data
+
+source("./prep_data/prep_functions.R")
+
+###### 0. Create folders to save the data -----------------
+
+# Directory to keep raw zipped files
+dir.create("./federal_roads_accidents")
+
+year <- 2020
+state <- NULL
+
+#### 1. Download and clean data, given a year and possibly a state -----------------
+
+get_raw_accidents <- function(reference_year,
+                              reference_state = "all") {
+
+  html_content <-
+    read_html("https://portal.prf.gov.br/dados-abertos-acidentes") %>%
+    html_node("#acontent ul") %>%
+    html_nodes("li")
+
+  data <- tibble(
+    href = {
+      html_content %>%
+        html_children %>%
+        html_attr("href")
+      },
+    year = {
+      html_content %>%
+        html_text() %>%
+        as.numeric()
+      }
+    ) %>%
+    filter(year == reference_year) %>%
+    mutate(href = glue("curl {href}/download | funzip")) %>%
+    pull(href) %>%
+    fread(encoding = "Latin-1") %>%
+    as_tibble()
+
+  if(reference_state == "all") {
+    reference_state <-
+      data %>%
+      pull(uf) %>%
+      unique()
+  }
+
+   data %<>%
+     filter(uf %in% reference_state) %>%
+     mutate(latitude = parse_double(latitude, locale = locale(decimal_mark = ',')),
+            longitude = parse_double(longitude, locale = locale(decimal_mark = ',')),
+            km = parse_double(km, locale = locale(decimal_mark = ',')),
+            id = as.character(id),
+            br = as.character(br))
+
+    data
+
+}
+
+
+accidents <- get_raw_accidents(year)
```

Comment:
Oi Pedro. Quando eu rodo essa linha 85 eu recebo o seguinte erro.

>  Error: object 'uf' not found
Run `rlang::last_error()` to see where the error occurred.